imap_upstream.adoc

```diff
@@ -161,6 +161,8 @@ zmprov mcf +zimbraReverseProxyUpstreamImapServers <server1> \
   +zimbraReverseProxyUpstreamImapServers <server2> \
  ...
 ----
+[NOTE]
+In an upgrade scenario values MUST BE assigned to `zimbraReverseProxyUpstreamImapServers` at the server level until all mailbox servers have been updated.
```

Comment:
I assume you mean "must NOT be"

Reply:
Actually I meant what was written based on the comment on ticket: ZCS-2402 which I have copied below for reference:

>>NOTE: Important to document that, in an upgrade scenario, the user must assign values to >>zimbraReverseProxyUpstreamImapServers at the server level, at least until all mailbox >>servers have been updated.

Is this incorrect?

Reply:
IIRC, the **MUST BE** is correct because the required soap calls are only supported on the upgraded mailbox servers. However, your question raises a concern about the vagueness of this note...

In my mind, one thing that is missing here is the answer to *WHY?*. While we might not want to have to answer that question, if we don't explain in more detail why this note exists we'll just add to the confusion because one needs to know *which server(s)* to add the `zimbraReverseProxyUpstreamImapServers` attribute/setting to and when.  It might help to have the document give a diagram of flow of traffic / logic and how these keys come in to play.

Suggestion: can we make these phrases bold in this NOTE:
- **MUST BE**
- **server level**

Also perhaps we should change "updated" to "upgraded"?

Reply:
@gordyt @fatkudu and I spoke about this during our daily today.  We decided that this did not cover all scenarios which would be encountered and therefore was insufficient.
I am going to open a new ticket and detail what documentation is needed in respect to the different install and upgrade scenarios that might be encountered in the field.

Reply:
@tmclane Let's just close this PR and resubmit updated PR based on new ticket that covers the factors we discussed this morning.